middle_end/flambda2/simplify/flow/flow_analysis.ml

```diff
@@ -46,7 +46,7 @@ let print_graph ~print ~print_name ~lazy_ppf ~graph =
   | Some print_name ->
     incr dot_count;
     let ppf = Lazy.force lazy_ppf in
-    print ~ctx:!dot_count ~print_name ppf graph
+    print ~ctx:!dot_count ~print_name ppf graph [@nontail]
```

Comment:
I'm finding this `[@nontail]` attribute strange... Is it needed because the function in inlined and in some cases the `print` function is overapplied?

Reply:
I'm also surprised: to my knowledge, `flow_analysis` does not use any `local` (at least not explicitly), so I would expect not to need such annotations.

Reply:
I guess the question is why is this additional closure locally allocated ?

Reply:
@stedolan ping on that question: why are some closures automatically allocated on the local stack even though there are no user annotations/requests to do so ?
I've see that this also applies to other things, such as floats, and this can be a problem sometimes, particularly in classic mode where it prevents any unboxing from taking place (see https://github.com/ocaml-flambda/flambda-backend/issues/1491#issuecomment-1662412627 ).

Maybe turning every regular heap allocations into a local allocations without any annotation is not that good an idea when it can prevent other optimisations, and even break tail calls. As an author it seems a bit like the compiler tries to break my code.